### DISCUSSION TITLE
Enhancement for sink config to refer to (the resolved) global timezone name

### DISCUSSION THREAD

**Discussion by klondikedragon** [vectordotdev/vector#19685] [General]
I'm using the http sink. I want to let the server know the timezone name (e.g., America/Chicago) that is configured for this vector system (the value of the global timezone configuration entry, or if the timezone is configured to "local" ideally vector would get and use the actual timezone name (e.g., America/Chicago) from the host OS).

This can help the server do the right thing if the server is parsing/processing unstructured textual log data that it receives, and it needs to know how to treat text timestamps that it's detecting/parsing if they do not have an explicit timezone. Right now vector configuration can refer to environment variables or secrets, but it doesn't appear it can refer to global configuration values. Is it currently possible for a sink configuration value (e.g., the http sink uri or header) to refer to the value of the timezone?

Possible ways this could be implemented:
- A special "environment variable" (name of TIMEZONE or maybe more unique like VECTOR.TIMEZONE) could be automatically added after global config is resolved that has the dynamic value of the global timezone configuration (including trying to resolve `TimeZone::Local` to the actual local timezone name). This does overload environment variable interpolation with a special value, so maybe it's not as clean an approach as would be desired. On the other hand, setting an actual environment variable could be beneficial for other places, like the exec source. That could get messy after a while if a lot of configuration values became set environment variables...
- Alternatively, a new type of setting interpolation could be added just like secret interpolation, e.g., `CONFIG[timezone]` -- on quick inspection, this would probably require a lot more code changes than just adding a special environment variable with the resolved timezone name.
- Alternatively, it would be easy enough to develop a custom sink that wrapped the http sink and directly used/transformed the global `cx.globals.timezone` into a request header, but that is less general, and it feels like other vector users might also want to send what the locally configured timezone is to a sink.

**Comment by jszwedko** [comment]
This is interesting! One additional way to implement this would be to add a VRL function for fetching it where it could be added to the event metadata (like `get_vector_timezone()`).

**Reply by klondikedragon** [reply]
That certainly is an option and feels pretty clean compared to overloading the environment variable config interpolation. It's a little less convenient but not overly so. One drawback of the VRL approach is that it would add data to each event, which doesn't quite match, as this is really a property of the sending system rather than any particular event, and the receiving system likely doesn't need to use/store the information for every event, it only needs it as a "default timezone" etc.

Any preference what direction you'd like to see? Depending on complexity I might have time to create a PR for consideration. Personally I might lean towards a new interpolation type (e.g., `CONFIG[timezone]`) but open to any option.

**Reply by jszwedko** [reply]
Thanks for the additional thoughts! I agree that, because it is static, there may be better ways to model it than attaching the metadata to each event. However, that is the existing pattern we have for this sort of thing (e.g. `get_hostname` is similar in that is generally static) so I think you'd have an easier time contributing a new VRL function for this than adding a new config feature (which would require more design thought and back-and-forth). It would still leave the door open to adding that sort of static config injection in the future.

**Reply by klondikedragon** [reply]
Makes sense. I've opened https://github.com/vectordotdev/vector/issues/19716 to track the enhancement to add a VRL function.

**Reply by klondikedragon** [reply]
This was implemented in https://github.com/vectordotdev/vector/pull/19727, thanks for being receptive to the idea.
